lib/util/Components.js

```diff
@@ -63,14 +63,50 @@ Components.prototype.get = function(node) {
  * @param {Object} props Additional properties to add to the component.
  */
 Components.prototype.set = function(node, props) {
+  const popsAreEquivalent = (propA, propB) => {
+    if (propA.name === propB.name) {
+      if (!propA.allNames && !propB.allNames) {
+        return true;
+      } else if (propA.allNames.join('') === propB.allNames.join('')) {
+        return true;
+      }
+      return false;
+    }
+    return false;
+  };
+
+  const merge = (propsList, newPropList) => {
+    const propsToAdd = [];
+    newPropList.forEach(newProp => {
+      let newPropisAlreadyInTheList = false;
+      propsList.some(prop => {
```

Comment:
The use of `some` here looks strange. `some()` returns a boolean but it is not used.

Something like this should work too ( didn't try it though ):
`const newPropIsAlreadyInTheList = propsList.some(prop => popsAreEquivalent(prop, newProp)`

I think the `I` in `newPropisAlreadyInTheList` should be capitalised.